### PR TITLE
setup: Clarify when local configuration is written

### LINKED ISSUE #159: `git-hub setup` should be global by default, if no global config exists
Having experienced https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=799286, I propose that `git-hub setup` default to `--global` if there's no global oauth token set; and otherwise it should operate on the local repository by default. 

I'm assuming that most common use case for this awesome tool is where a global config is necessary.

I can see that `setup --global` is similar in design to `git config --global`, but this may be jarring.

:+1:

### PR DESCRIPTION
**Pull request by llucax**
Also update man page to be more clear about where and how the
configuration is stored.

Fixes #159.

### CODE REVIEW

File: man.rst [hunk 1]
```diff
@@ -55,11 +55,15 @@ COMMANDS
...
 `setup`
   This command performs an initial setup to connect to GitHub. It basically
-  asks GitHub for an authorization token and stores it in the configuration
+  asks GitHub for an authorization token and stores it in the Git configuration
   variable `hub.oauthtoken` for future use so you don't need to type your
   password each time (or store it in the config). The username is also stored
   for future use in the `hub.username` variable. If the base URL is specified,
-  it is stored in `hub.baseurl` too.
+  it is stored in `hub.baseurl` too. By default configuration is stored in the
+  repository's ``.git/config`` file (using ``git config``). If you want your
```
Comment (guyhughes) [inline man.rst]: :+1:

Comment (salewski) [inline man.rst]: :+1:

**Comment by leandro-lucarella-sociomantic** [comment]
This is kind of in the middle of a new feature and a bug fix, but I will put it in v0.10.2 because the impact should be low enough to be brave about it (last famous words).

**Comment by mihails-strasuns-sociomantic** [comment]
LGTM

**Comment by mihails-strasuns-sociomantic** [comment]
This pull request has been rebased via `git hub pull rebase`. Original pull request HEAD was fe426ea415fd0e6b84fcdc7d653c1121a679970e, new (rebased) HEAD is a09f8ac6ef69e084595a155d6cb7c5da544115e4

**Comment by leandro-lucarella-sociomantic** [comment]
Arrgh, I messed up, a few of the PRs in here were done against master instead of v0.10.x. I will not include this one in v0.10.2 because it's in master and technically is not a bug fix.

**Pull request closed without merging**